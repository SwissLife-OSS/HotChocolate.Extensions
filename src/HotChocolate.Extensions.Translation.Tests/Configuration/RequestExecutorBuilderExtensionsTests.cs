using System.Threading.Tasks;
using FluentAssertions;
using HotChocolate.Execution;
using HotChocolate.Execution.Configuration;
using Microsoft.Extensions.DependencyInjection;
using Snapshooter.Xunit;
using Xunit;

namespace HotChocolate.Extensions.Translation.Tests.Configuration
{
    public class RequestExecutorBuilderExtensionsTests
    {
        public class Query
        {
            public string Foo { get; } = "bar";
            public DummyEnum Bar { get; } = DummyEnum.Qux;
        }

        public enum DummyEnum
        {
            Qux
        }

        [Fact]
        public async Task AddTranslation_WithTranslatedEnum_ShouldAddTheTypeForTheTranslatedEnum()
        {
            //Arrange
            var services = new ServiceCollection();
            IRequestExecutorBuilder builder = services
                .AddGraphQLServer()
                .AddQueryType<Query>(d =>
                {
                    d.Field(q => q.Bar).Translate<DummyEnum>("translation_path");
                });

            //Act
            builder = builder
                .AddTranslation(c => c
                    .AddTranslatableType<DummyEnum>()
                );

            //Assert
            ISchema schema = await builder.BuildSchemaAsync();
            schema.ToString().Should().MatchSnapshot();
        }

        [Fact]
        public async Task AddTranslation_WithCustomNameForInterface_ShouldAddTheInterfaceWithTheCorrectName()
        {
            //Arrange
            var services = new ServiceCollection();
            IRequestExecutorBuilder builder = services
                .AddGraphQLServer()
                .AddQueryType<Query>();

            //Act
            builder = builder
                .AddTranslation(c => c
                    .SetTranslationInterfaceName("MyTranslationInterfaceType")
                );

            //Assert
            ISchema schema = await builder.BuildSchemaAsync();
            schema.ToString().Should().MatchSnapshot();
        }
    }
}
