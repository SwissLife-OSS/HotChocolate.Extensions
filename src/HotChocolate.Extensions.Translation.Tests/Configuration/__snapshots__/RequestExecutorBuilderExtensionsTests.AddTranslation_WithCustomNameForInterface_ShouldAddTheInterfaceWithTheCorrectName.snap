schema {
  query: Query
}

interface MyTranslationInterfaceType {
  label: String!
}

type Query {
  foo: String!
  bar: DummyEnum!
}

enum DummyEnum {
  QUX
}
